### ISSUE TITLE
linter: `unicorn/prefer-node-protocol` false positive for internal constants

### ISSUE THREAD

**Issue by elusiveunit** [oxc-project/oxc#25545]
### What version of Oxlint are you using?

1.78.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```typescript
export default defineConfig({
	plugins: ['unicorn'],
	rules: {
		'unicorn/prefer-node-protocol': 'warn',
	},
});
```

### What happened?

I guess this could be considered a user problem since it's a project specific configuration, but it would be nice to have it working like eslint-plugin-unicorn.

constants/url.ts
```typescript
export const ADMIN_PATH = '/admin';
```

tsconfig.json
```jsonc
{
	// [...]
	"compilerOptions": {
		"paths": {
			"constants/*": ["./src/constants/*"],
		}
	}
}
```

example.ts
```typescript
import { randomUUID } from 'crypto';
import { ADMIN_PATH } from 'constants/url';

console.log(ADMIN_PATH, randomUUID());
```

output from `npx oxlint`:
```
  ⚠ unicorn(prefer-node-protocol): Prefer using the `node:` protocol when importing Node.js built-in modules.
   ╭─[example.ts:1:28]
 1 │ import { randomUUID } from 'crypto';
   ·                            ────────
 2 │ import { ADMIN_PATH } from 'constants/url';
   ╰────
  help: Prefer `node:crypto` over `crypto`.

  ⚠ unicorn(prefer-node-protocol): Prefer using the `node:` protocol when importing Node.js built-in modules.
   ╭─[example.ts:2:28]
 1 │ import { randomUUID } from 'crypto';
 2 │ import { ADMIN_PATH } from 'constants/url';
   ·                            ───────────────
 3 │
   ╰────
  help: Prefer `node:constants/url` over `constants/url`.
```

The first one is valid and matches the behavior of upstream:
```
example.ts
  1:28  warning  Prefer `node:crypto` over `crypto`  unicorn/prefer-node-protocol
```

The second happens only with oxlint and the suggested fix is an invalid import, `Cannot find module 'node:constants/url' or its corresponding type declarations. ts(2307)`.

**Comment by elusiveunit** [comment]
For reference, the upstream logic:

* https://github.com/sindresorhus/eslint-plugin-unicorn/blob/398606c58964ecb074321e5bfa1f4181456f4a4d/rules/prefer-node-protocol.js#L54
* https://github.com/sindresorhus/builtin-modules/blob/main/builtin-modules.json
